File: app/src/test/java/com/example/adiphong/rx2/CompositeDisposableTest.java
package com.example.adiphong.rx2;

import org.junit.Test;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Observable;
import io.reactivex.Single;
import io.reactivex.disposables.CompositeDisposable;

/**
 * Created by adiphong on 27/11/2017 AD.
 */

public class CompositeDisposableTest {
    CompositeDisposable disposable = new CompositeDisposable();

    //just
    //////////////////////////////////////////////////////////////////////////////////////////////////
    @Test
    public void flowableJust() {
        disposable.add(Flowable
                .just("Hello", "World")
                .subscribe(
                        s -> System.out.println("---------------flowable  result " + s)
                )
        );
    }

    @Test
    public void observableJust() {
        disposable.add(Observable
                .just("Hello", "World")
                .subscribe(
                        s -> System.out.println("---------------observable result " + s)
                )
        );
    }

    @Test
    public void maybeJust() {
        // can be only one data
        disposable.add(Maybe.just("Hello")
                .subscribe(
                        s -> System.out.println("---------------maybe result " + s)
                )
        );
    }

    @Test
    public void singleJust() {
        // can be only one data
        disposable.add(Single.just("Hello")
                .subscribe(
                        s -> System.out.println("---------------single result " + s)
                )
        );
    }

    @Test
    public void completeAbleJust() {
        //can be only one data
        disposable.add(Completable.create(emitter -> {
                    emitter.onComplete();
                })
                        .subscribe(
                                () -> System.out.println("--------- completableCreate onComplete "),
                                throwable -> System.out.println("--------- completableCreate onError")
                        )
        );
    }

}

File: app/src/test/java/com/example/adiphong/rx2/CompositeDisposableTest.kt
package com.example.adiphong.rx2

import org.junit.Test

import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Maybe
import io.reactivex.Observable
import io.reactivex.Single
import io.reactivex.disposables.CompositeDisposable

/**
 * Created by adiphong on 27/11/2017 AD.
 */

class CompositeDisposableTest {
    private var disposable = CompositeDisposable()

    //just
    //////////////////////////////////////////////////////////////////////////////////////////////////
    @Test
    fun flowableJust() {
        disposable.add(Flowable
                .just("Hello", "World")
                .subscribe { s -> println("---------------flowable  result $s") }
        )
    }

    @Test
    fun observableJust() {
        disposable.add(Observable
                .just("Hello", "World")
                .subscribe { s -> println("---------------observable result $s") }
        )
    }

    @Test
    fun maybeJust() {
        // can be only one data
        disposable.add(Maybe.just("Hello")
                .subscribe { s -> println("---------------maybe result $s") }
        )
    }

    @Test
    fun singleJust() {
        // can be only one data
        disposable.add(Single.just("Hello")
                .subscribe { s -> println("---------------single result $s") }
        )
    }

    @Test
    fun completeAbleJust() {
        //can be only one data
        disposable.add(Completable.create { emitter -> emitter.onComplete() }
                .subscribe(
                        { println("--------- completableCreate onComplete ") },
                        { println("--------- completableCreate onError") }
                )
        )
    }
}
